contiamo/tests/test_requests.py

```diff
@@ -107,3 +107,8 @@ def test_two_arguments(self):
         df = pd.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]})
         with pytest.raises(InvalidRequestError):
             data_client.upload(dataframe=df, filename=utils.file_test_data('mock_data.csv'))
+
+    def test_invalid_chunk_size(self):
```

Comment:
this test is unnecessary IMO. A test for the chunking / data slicing is probably the only new functionality that needs to be tested (not saying you have to, just if you wanted to add more tests).

Reply:
It's actually a regression test in disguise: I just want to make sure no-one removes the functionality and hence the `chunk_size` parameter by mistake (in which case a `TypeError` would be raised).

Reply:
I didn't add a proper test because it was a bit impractical, but I've thought of a couple of ways to do so over lunch...

Reply:
>It's actually a regression test in disguise: I just want to make sure no-one removes the functionality and hence the chunk_size parameter by mistake (in which case a TypeError would be raised).

fair enough 

>I didn't add a proper test because it was a bit impractical, but I've thought of a couple of ways to do so over lunch... What do you think?

yeah, I thought it might be more trouble than it's worth, so happy to leave this decision to you :)